src/binmode/legacy4third.c

```diff
@@ -75,7 +75,23 @@ static uint32_t remain_bytes;
 static bool set_aux_pins = true;
 static bool hold_value = true;
 static bool wp_value = true;
-static uint8_t binmode_debug = 0; // Debug mode flag
+
+// For Atmel parts which have a flash size > 64Kbytes, an additional
+// command is needed - the "Extended High Byte" address write.  This
+// bool will be set when avrdude first queries the part, and will be
+// used later if required.
+static bool req_EHB_write = false; // This part has > 64K flash
```

Comment:
I have some concern about this hack.  It's a hidden (from client and user) state variable.
Previously, it was automatically reset to `false` on subsequent `Read Signature Byte` (`0x30`) commands.

Now, it will never be set to `false` after anytime that it gets detected.   I will comment below where this might be updated to continue to have more automatic update.